### DISCUSSION TITLE
How to toggle on/off automatic parent-children visual groupings?

### DISCUSSION THREAD

**Discussion by shmup** [piroor/treestyletab#3295] [Q&A]
I set up this current behavior a long time ago when I didn't want it to automatically create a parent-child visual when opening a link from a page.

Now I want to revert this behavior but get lost in the Options, and my few tests aren't working.

Currently: if I manually create a child-parent by dragging a tab onto another, then any tabs I open from that parent page will render as a child (cool). But if I don't create this manual relationship, they all open inline (uncool).

I attached a .gif to show what I mean.

![022223-131847-KPk-firefox](https://user-images.githubusercontent.com/118710/220720419-abfe1c17-7035-453f-918c-92901c805b17.gif)

Thanks, and I'll update this if I find the correct answer myself.

**Comment by irvinm** [comment]
Can you check this option?  

![image](https://user-images.githubusercontent.com/979729/221295031-b8a787bf-b9ea-498a-93a8-1ad5a91d20b6.png)

**Reply by shmup** [reply]
Looks like I did have `a child of the parent tab` selected.

The only thing different is my insert position, but I changed it to match yours anyways.

I'll look around some more to see if anything can conflict?

**Comment by piroor** [comment]
TST has ability to fix up tree structure for inserted tabs. For example, assume that there are three tabs:

* A (parent of B, active)
  * B (child of A)
* C (independent)

If you configured Firefox to open new tab next the active tab and you try to open a new tab from a link in the tab A, you'll get a new tab D as:

* A (parent of B, active)
* D (independent, opened from A)
  * B (child of A)
* C (independent)

On such situation TST attaches D to the tab A as its new first child based on its position. Then tabs will be:

* A (parent of B and D, active)
  * D (child of A, opened from A)
  * B (child of A)
* C (independent)

Please remind that the auto-fix feature works only when there is any "broken" tree structure. Such an auto-fix won't be triggered if you open a new tab E from an independent tab C like:

* A (parent of B and D)
  * D (child of A, opened from A)
  * B (child of A)
* C (independent, active)
* E (independent, opened from C)

On such case TST keep the tab E as independent.

I think this feature can describe what you saw.

**Reply by shmup** [reply]
Ultimately, after I noticed there are reset options at the bottom of the preferences, I took the easy way out. Mostly I was curious if that would work, because I was really rather confused!

It all works, and I think the defaults are pretty sane. So I'm glad I can forget about this now haha.

Thanks for taking the time to write about tab insertion behavior though.

Not entirely sure how to mark this discussion, maybe it can be labeled `Mu`
